### PR TITLE
chore: Convert Standard to eslint-config-standard

### PR DESCRIPTION
**Pull request by nschonni**
### Why:

Standard is a wrapper of opionated ESLint rules. By using the direct ESLint config, there is better integration with IDEs for showing ESLint errors while maintaining the existing style preference. Also can drop some of the JSX related dependencies that are bundled as part of the larger `standard` package.

### What's being changed:

- Remove the `standard` package and replace with the ESLint packages for `standard`
- Fix issues flagged from going from older version of standard to latest ESLint+Standard
- Update calls to standard in the CI to use ESLint

### Check off the following:
- [x] All of the tests are passing.
- [ ] I have reviewed my changes in staging.
- [ ] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [ ] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -146,13 +150,6 @@
     "node": "12 - 14"
   },
   "repository": "https://github.com/github/docs",
-  "standard": {
```
Comment (heiskr) [inline package.json:149]: I'm curious about thoughts on including the eslint config and ignore in package.json instead of as separate files.

**Review by heiskr — APPROVED** [review]
This looks like a good change to me and one that I've run into personally.

File: package.json (hunk 1 above)
Comment (nschonni) [reply package.json:149]: I believe the separate ignore file usually has better performance, but I've not looked at using the inline config in package.json before

**Comment by chiedo** [comment]
@nschonni thanks for creating this! Can you resolve the conflicts 🙏🏿

**Comment by nschonni** [comment]
@chiedo done!

**Comment by chiedo** [comment]
Just want one more review on this one! @JasonEtco can you please review!

**Review by JasonEtco — APPROVED** [review]
This looks great, thanks for all the improvements here (and everywhere else) @nschonni!

**Pull request merged**